standard.go

```diff
@@ -0,0 +1,23 @@
+package retryablehttp
+
+import "net/http"
+
+// NewStandardClient creates a new StandardClient with a Client with default settings.
+func NewStandardClient() StandardClient {
```

Comment:
I think this should be returning a `*StandardClient` considering that's what the `Do()` method operates on.